src/services/EdFi.DataManagementService.Api/Core/Model/DescriptorDocument.cs

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Diagnostics;
+using System.Text.Json.Nodes;
+
+namespace EdFi.DataManagementService.Api.Core.Model;
+
+/// <summary>
+/// The form of all descriptor documents
+/// </summary>
+public class DescriptorDocument(JsonNode _document)
+{
+    private static readonly DocumentObjectKey _descriptorKey = new("descriptor");
+
+    /// <summary>
+    /// Creates a new DocumentIdentity from the DescriptorDocument
+    /// </summary>
+    public DocumentIdentity ToDocumentIdentity()
+    {
+        string? namespaceName = _document["namespace"]?.GetValue<string>();
+        Debug.Assert(namespaceName != null, "Failed getting namespace field, JSON schema validation?");
```

Comment:
It actually works as I'd expect. I think the Microsoft documentation needs updating because it talks about message boxes at the top, but later that it's only "When the application runs in user interface mode...". Example output:
`---- DEBUG ASSERTION FAILED ----
---- Assert Short Message ----
This test of Debug.Assert() should fail
---- Assert Long Message ----

   at EdFi.DataManagementService.Api.Core.Middleware.ExtractDocumentInfoMiddleware.Execute(PipelineContext context, Func`1 next) in /home/brad/work/tanager/Data-Management-Service/src/services/EdFi.DataManagementService.Api/Core/Middleware/ExtractDocumentInfoMiddleware.cs:line 26`

The great thing about it is that code analysis understands the assertion, so if you assert != null on a nullable it knows it's not null after the assertion. This is just like the behavior of the TypeScript-aware `invariant` library we used in Meadowlark.

Reply:
Do we need to throw an `InvalidOperationException` if the `namespace is null`?